src/oneD/IonFlow.cpp

```diff
@@ -57,6 +58,23 @@ IonFlow::IonFlow(IdealGasPhase* ph, size_t nsp, size_t points) :
     m_do_electric_field.resize(m_points,false);
 }
 
+IonFlow::IonFlow(shared_ptr<Solution> sol, size_t nsp, size_t points) :
+    IonFlow(sol->thermo().get(), nsp, points)
+{
+    m_solution = sol;
+    m_kin = m_solution->kinetics().get();
+    m_trans_shared = m_solution->transport();
+    m_trans = m_trans_shared.get();
```

Comment:
If I remember correctly, it is needed to prevent hard crashes. There are instances where the transport object would get destroyed if the shared pointer isn’t created here.

Reply:
Oh, I see it now -- if you call `setTransportModel` on the `Solution` object, then that would cause the current object to be deleted unless the `StFlow` object is also holding a copy of the `shared_ptr`. Arguably, the same thing is true for the `Kinetics` object and the use of the `Solution::setKinetics` method, though I don't think there's any case where we currently use the ability to replace the `Kinetics` object that way.

Reply:
Your interpretation is spot-on. There’s no good way around this without replacing raw pointers, which is well beyond of what I was trying to do in this PR.

Reply:
I updated the docstring of `StFlow::m_trans_shared` to clarify.